src/jgikbase/idmapping/core/errors.py

```diff
@@ -45,6 +45,9 @@ class ErrorType(Enum):
     NO_SUCH_NAMESPACE =      (50010, "No such namespace")  # noqa: E222 @IgnorePep8
     """ There is no namespace with the specified name. """
 
+    NO_SUCH_AUTHSOURCE =     (50020, "No such authentication source")  # noqa: E222 @IgnorePep8
```

Comment:
This is what I meant to ask before: why assign numbers like this to each error -- what do you gain? Why not give it an identifier like "no_such_authsource"? Maybe it allows you to do something like `error_code >= 30000 and error_code < 4000` to identify that it's in a group?

Reply:
The latter, and also that it's pretty standard practice to provide an error type ID - mongo does this, for example. I provide both and ID and a text description so that a client programmer can use whichever they prefer.